src/main/scala/com/geirsson/codegen/Codegen.scala

```diff
@@ -54,62 +54,73 @@ case class Codegen(options: CodegenOptions, namingStrategy: NamingStrategy) {
   val excludedTables = options.excludedTables.toSet
   val columnType2scalaType = options.typeMap.pairs.toMap
 
+  def results(resultSet: ResultSet): Iterator[ResultSet] = {
+    new Iterator[ResultSet] {
+      def hasNext = resultSet.next()
+      def next() = resultSet
+    }
+  }
+
   def getForeignKeys(db: Connection): Set[ForeignKey] = {
-    val sb = Set.newBuilder[ForeignKey]
     val foreignKeys =
       db.getMetaData.getExportedKeys(null, options.schema, null)
-    while (foreignKeys.next()) {
-      sb += ForeignKey(
+    results(foreignKeys).map { row =>
+      ForeignKey(
         from = SimpleColumn(
-          tableName = foreignKeys.getString(FK_TABLE_NAME),
-          columnName = foreignKeys.getString(FK_COLUMN_NAME)
+          tableName = row.getString(FK_TABLE_NAME),
+          columnName = row.getString(FK_COLUMN_NAME)
         ),
         to = SimpleColumn(
-          tableName = foreignKeys.getString(PK_TABLE_NAME),
-          columnName = foreignKeys.getString(PK_COLUMN_NAME)
+          tableName = row.getString(PK_TABLE_NAME),
+          columnName = row.getString(PK_COLUMN_NAME)
         )
       )
-    }
-    sb.result()
+    }.toSet
   }
 
   def getTables(db: Connection, foreignKeys: Set[ForeignKey]): Seq[Table] = {
-    val sb = Seq.newBuilder[Table]
     val rs: ResultSet =
       db.getMetaData.getTables(null, options.schema, "%", Array("TABLE"))
-    while (rs.next()) {
-      if (!excludedTables.contains(rs.getString(TABLE_NAME))) {
-        val name = rs.getString(TABLE_NAME)
-        sb += Table(
+    results(rs).flatMap { row =>
+      val name = row.getString(TABLE_NAME)
+      if (!excludedTables.contains(name)) {
+        val columns = getColumns(db, name, foreignKeys)
+        val mappedColumns = columns.filter(_.isRight).map(_.right.get)
+        val unmappedColumns = columns.filter(_.isLeft).map(_.left.get)
+        if (!unmappedColumns.isEmpty)
+          println(s"The following columns from table $name need a mapping: $unmappedColumns")
```

Comment:
Maybe turn `println` into a `warn`?

```scala
def warn(msg: String): Unit = {
  System.err.println(s"[${Console.YELLOW}warn${Console.RESET}] $msg")
}
```

I would like stdout to only be the generated source code for use in pipes, etc.